torchsynth/synth.py

```diff
@@ -214,15 +214,25 @@ def randomize(self, seed: Optional[int] = None):
         Randomize all parameters
         """
         if seed is not None:
-            # Profile to make sure this isn't too slow?
-            mt19937_gen = csprng.create_mt19937_generator(seed)
+            self.generator.manual_seed(seed)
             for parameter in self.parameters():
                 if not ModuleParameter.is_parameter_frozen(parameter):
-                    parameter.data.uniform_(0, 1, generator=mt19937_gen)
+                    # TODO figure out how to support different batch sizes and
+                    # maintain reproducibility.
+                    if self.device.type != "cpu":  # pragma: no cover
```

Comment:
A little cleaner to be like self.device.type != self.generator.type OR just make self.generator a local variable and call it cpu_rng and be done with it